src/data_structure/versions/translate_heatrate_parameters.jl

```diff
@@ -0,0 +1,99 @@
+		"entity_class_name" => "unit__node__node", "name" => "fix_units_on_coefficient_in_out")
+	)
+	run_request(db_url, "call_method", ("add_parameter_definition_item",), Dict(
+		"entity_class_name" => "unit__node__node", "name" => "fix_ratio_in_out_unit_flow")
+	)
+	# Get entities in the class and get all alternatives
+	entity_items = run_request(db_url, "call_method", ("get_entity_items",), Dict(
+		"entity_class_name" => "unit__node__node")
+	)
+	alternative_items = run_request(db_url, "call_method", ("get_alternative_items",))
+	for entity in entity_items
+		for alternative in alternative_items
+			# Get the value of the old unit_idle_heat_rate parameter
+			pval = run_request(db_url, "call_method", ("get_parameter_value_item",), Dict(
+				"entity_class_name" => "unit__node__node", 
+				"parameter_definition_name" => "unit_idle_heat_rate", 
+				"entity_byname" => (entity["element_name_list"]), 
+				"alternative_name" => alternative["name"])
+			)
+			if length(pval) > 0
+				# Add the value as fix_units_on_coefficient_in_out into the database
+				run_request(
+					db_url, "call_method", ("add_update_parameter_value_item",), Dict(
+						"entity_class_name" => "unit__node__node", 
+						"parameter_definition_name" => "fix_units_on_coefficient_in_out", 
+						"entity_byname" => (entity["element_name_list"]), 
+						"alternative_name" => alternative["name"], 
+						"value" => pval["value"], 
+						"type" => pval["type"])
+				)
+			end
+			# Get the value of the old unit_incremental_heat_rate parameter
+			pval = run_request(db_url, "call_method", ("get_parameter_value_item",), Dict(
+				"entity_class_name" => "unit__node__node", 
+				"parameter_definition_name" => "unit_incremental_heat_rate", 
+				"entity_byname" => (entity["element_name_list"]), 
+				"alternative_name" => alternative["name"])
+			)
+			if length(pval) > 0
+				# Add the value as fix_ratio_in_out_unit_flow into the database
+				run_request(
+					db_url, "call_method", ("add_update_parameter_value_item",), Dict(
+						"entity_class_name" => "unit__node__node", 
+						"parameter_definition_name" => "fix_ratio_in_out_unit_flow", 
+						"entity_byname" => (entity["element_name_list"]), 
+						"alternative_name" => alternative["name"], 
+						"value" => pval["value"], 
+						"type" => pval["type"])
+				)
+			end
+		end
+	end
+	# Remove old parameter definition
+	pdef = run_request(db_url, "call_method", ("get_parameter_definition_item",), Dict(
+		"entity_class_name" => "unit__node__node", "name" => "unit_idle_heat_rate")
+	)
+	if length(pdef) > 0
+		run_request(db_url, "call_method", ("remove_parameter_definition_item", pdef["id"]))
+	end
+	pdef = run_request(db_url, "call_method", ("get_parameter_definition_item",), Dict(
+		"entity_class_name" => "unit__node__node", "name" => "unit_incremental_heat_rate")
+	)
+	if length(pdef) > 0
+		run_request(db_url, "call_method", ("remove_parameter_definition_item", pdef["id"]))
+	end
+
+	true
+end
```

Comment:
This migration function uses 0.8 spinedb_api so we cannot merge these changes into SpineOpt's 0.7 branch, directly - we'd need to write a new migration function using the corresponding 0.7 spinedb_api.

Reply:
I also just remembered there was some discussion about keep supporting the heat rate parameters but introduce a new prepocessing step to translate them into fix-ratio parameters. That would remove the need for the migration script but of course the question is do we want to keep supporting the heat rate parameters?